source/fab/steps/grab/fcm.py

```diff
@@ -3,23 +3,16 @@
 #  For further details please refer to the file COPYRIGHT
 #  which you should have received as part of this distribution
 # ##############################################################################
-from typing import Dict
+from fab.steps.grab.svn import SvnCheckout, SvnExport, SvnMerge
 
-from fab.steps.grab import GrabSourceBase
-from fab.tools import run_command
 
+class FcmExport(SvnExport):
+    command = 'fcm'
```

Comment:
I don't like this breaking of encapsulation. I would rather have the command passed to the constructor so it is explicit and encapsulation is respected.

i.e. The base Subversion class takes the subversion command as an argument on its constructor. The FCM constructor then calles `super().__init__('fcm')`.

Reply:
Absolutely we can do that. There's a point about build script readability but we can address that with helper functions instead.

Reply:
Are you sure class variables work like that?

A class variable is evaluated _on declaration_, before any instantiation takes place, when the class itself is being evaluated. It's shared between all instances.

It's easy to check if it behaves the way you're suggesting:

```
from fab.steps.grab.fcm import FcmCheckout
from fab.steps.grab.svn import SvnCheckout

f1 = FcmCheckout('', '')
s1 = SvnCheckout('', '')

f2 = FcmCheckout('', '')
s2 = SvnCheckout('', '')

print(f1.command)
print(f2.command)
print(s1.command)
print(s2.command)
print()
print(f1.command)
print(s1.command)
print(f2.command)
print(s2.command)
```
```
fcm
fcm
svn
svn

fcm
svn
fcm
svn
```
We easily know exactly which command is being called by each class. There's no ambiguity, no race condition. This is how class variables work, unless I'm having a huge brain fart - entirely possible!

Calling a constructor won't change the value of a class variable, even by accident:
```
class C(object):
    att = 1
    def __init__(self):
        self.att = 2

c = C()
print(c.att)
print(c.__class__.att)
```
```
2
1
```

I'm happy to switch to dependency injection for the command, no problem at all.
But if the request is potentially based on a misunderstanding of Python, I think it's also useful for us to explore that, just to be sure.